docs/reference/notifications.md

```diff
@@ -0,0 +1,302 @@
+
+```py
+class EmailBackend(Backend):
+
+    id = "email"
+
+    def send(self, notification: Notification, recipient: Recipient):
+        """Mail sending implementation."""
+        content = self.render_template(notification, recipient)
+        resp = send_email({
+            "subject": content["subject"],
+            "html_body": content["html_body"],
+            "plain_body": content["plain_body"],
+            "recipients": [f"{recipient.name} <{recipient.email}>"],
+            "sender": current_app.config["MAIL_DEFAULT_SENDER"],
+        })
+        return resp
+```
+
+## Templates
+
+The Jinja templates provided shall include all parts of the notification that are subject to special formatting by a backend (e.g. subject, HTML/plaintext/markdown body, etc.) in separate Jinja blocks.
+Additional backends should provide their own templates, to be as specific as possible.
+
+This is an example notification
+
+```jinja
+{# notifications/community-submission.submitted.jinja #}
+
+{%- block subject -%}
+New record submission for your community {{ notification.context.get("request").get("receiver").get("metadata").get("title") }} submitted by {{ notification.context.get("request").get("created_by").get("username") }}
+{%- endblock subject -%}
+
+{%- block html_body -%}
+<p>The record "{{ notification.context.get("request").get("topic").get("metadata").get("title") }}" was submitted to your community {{ notification.context.get("request").get("receiver").get("metadata").get("title") }} by {{ notification.context.get("request").get("created_by").get("username") }}.</p>
+
+<a href="{{ notification.context.get("request").get("links").get("self_html") }}" class="button">Review the request</a>
+{%- endblock html_body -%}
+
+{%- block plain_body -%}
+The record "{{ notification.context.get("request").get("topic").get("metadata").get("title") }}" was submitted to your community {{ notification.context.get("request").get("receiver").get("metadata").get("title") }} by {{ notification.context.get("request").get("created_by").get("username") }}.
+
+Review the request: {{ notification.context.get("request").get("links").get("self_html") }}
+{%- endblock plain_body -%}
+
+{# Markdown for Slack/Mattermost/chat #}
+{%- block md_body -%}
+The record "{{ notification.context.get("request").get("topic").get("metadata").get("title") }}" was submitted to your community {{ notification.context.get("request").get("receiver").get("metadata").get("title") }} by {{ notification.context.get("request").get("created_by").get("username")}}.
+
+[Review the request]({{ notification.context.get("request").get("links").get("self_html") }})
+{%- endblock md_body -%}
+```
+
+## Configuration Values
+
+Configuration values used in the `invenio-notifications` module can be overriden, in order to adapt instances to specific needs.
+
```

Comment:
Suggestion:
Email notifications lack port in links if it's not explicitly added in either config:
```python
SITE_UI_URL = "https://127.0.0.1:5000"
SITE_API_URL = "https://127.0.0.1:5000/api"
```
or cli (e.g. invenio-cli run --port 5000). The celery worker appears to struggle to read the current port value. Therefore, to me, it's a good idea if we document that specifying the port number is necessary for the proper functioning of notification links."